src/77480.c

```diff
@@ -783,7 +787,59 @@ void render_player(void) {
     }
 }
 
-INCLUDE_ASM(void, "77480", render_player_model);
+void render_player_model(void) {
+    RenderTask task;
+    RenderTask* rtPtr = &task;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 x, y, z;
+    s8 renderModeTemp;
+    void (*phi_a0)(void);
+
+    if (playerStatus->flags & PLAYER_STATUS_FLAGS_40000000) {
+        playerStatus->flags &= ~PLAYER_STATUS_FLAGS_40000000;
+        get_screen_coords(gCurrentCamID, playerStatus->position.x, playerStatus->position.y,
+                          playerStatus->position.z, &x, &y, &z);
+        if (!(playerStatus->flags & PLAYER_STATUS_FLAGS_20000)) {
+            if (playerStatus->alpha1 != playerStatus->alpha2) {
+                if (playerStatus->alpha1 < 254) {
+                    renderModeTemp = 22;
+                    if (!(playerStatus->animFlags & PLAYER_STATUS_ANIM_FLAGS_1000000)) {
+                        renderModeTemp = 17;
+                    }
+
+                    playerStatus->renderMode = renderModeTemp;
```

Comment:
are you sure renderModeTemp is needed? can you just do the payerStatus->renderMode= in both the if and else?

Reply:
doesn't match with if/else